starknet/docs/its.md

```diff
@@ -0,0 +1,574 @@
+                                     YES)
+  -p, --privateKey < privateKey >    private key for Starknet account(testnet
+                                     only, not required for offline tx
+                                     generation) (env: STARKNET_PRIVATE_KEY)
+  --accountAddress <accountAddress>  Starknet account address (env:
+                                     STARKNET_ACCOUNT_ADDRESS)
+  -h, --help                         display help for command
+```
+
+**Note:** Like ITS deployment, ITF deployment does not support offline mode or gas estimation as it uses direct contract deployment.
+
+**Example:**
+```bash
+npx ts-node starknet/its/deploy-itf.ts \
+  --env testnet \
+  --interchainTokenService 0x... \
+  --owner 0x... \
+  --salt 0x123 \
+  --privateKey 0x... \
+  --accountAddress 0x...
+```
+
+## Token Operations
+
+### Deploy Interchain Token
+
+Deploy a new interchain token locally via the InterchainTokenFactory.
+
+```
+Usage: deploy-token [options]
+
+Deploy a new interchain token on Starknet via InterchainTokenFactory
+
+Options:
+  --salt <salt>                      Salt for token deployment
+  --name <name>                      Token name
+  --symbol <symbol>                  Token symbol
+  --decimals <decimals>              Token decimals
+  --initialSupply <amount>           Initial supply to mint (defaults to 0)
+  --minter <address>                 Minter address (if initial supply > 0,
+                                     mintership will be transferred to this
+                                     address)
+  -e, --env <env>                    environment (choices: "devnet-amplifier",
+                                     "mainnet", "stagenet", "testnet", default:
+                                     "testnet", env: ENV)
+  -y, --yes                          skip deployment prompt confirmation (env:
+                                     YES)
+  -p, --privateKey < privateKey >    private key for Starknet account(testnet
+                                     only, not required for offline tx
+                                     generation) (env: STARKNET_PRIVATE_KEY)
+  --accountAddress <accountAddress>  Starknet account address (env:
+                                     STARKNET_ACCOUNT_ADDRESS)
+  -h, --help                         display help for command
+
+Note: This command deploys tokens locally via the InterchainTokenFactory.
+- If initialSupply > 0, tokens will be minted to the deployer and mintership transferred to the minter
+- If initialSupply = 0, the minter address will be set directly
+- For cross-chain deployment, use a separate command after local deployment
+```
+
+**Examples:**
+
+Deploy with initial supply:
+```bash
+npx ts-node starknet/its/deploy-token.ts --salt my-token-salt --name "My Token" --symbol "MTK" --decimals 18 --initialSupply 1000000 --minter 0x123... --privateKey 0x... --accountAddress 0x...
+```
+
+Deploy without initial supply (minter gets set directly):
```

Comment:
```suggestion
Deploy without initial supply (minter is required in this case):
```
transferred or set directly is an implementation detail, I think we don't need to mention it here. What matters is that if the initial supply is not set, a minter has to be passed otherwise it'll revert.

Reply:
I also added [a check](https://github.com/eigerco/axelar-contract-deployments/pull/12/commits/85e4aa2a2271a65699f4a0160003ba62e21208c9) for it